src/nm_mon_daemon.c

```diff
@@ -44,27 +44,30 @@ typedef struct nm_clean_data {
     { NM_MON_VMS_INIT, NULL, NULL, NULL, NM_THR_CTRL_INIT, NM_THR_CTRL_INIT }
 
 #if defined (NM_OS_LINUX)
-static void nm_mon_cleanup(int rc, void *arg)
+static nm_clean_data_t *clean_ptr = NULL;
+
+static void nm_mon_cleanup(void)
 {
-    nm_clean_data_t *data = arg;
     const nm_cfg_t *cfg = nm_cfg_get();
 
-    nm_debug("mon daemon exited: %d\n", rc);
+	if (!clean_ptr) return;
```

Comment:
This should be done after nm_debug("mon daemon exited\n");
Please also rewrite as:
```
if (!clean_ptr)
    return;
```

Reply:
```
nm_debug("error 'clean_ptr' not set\n");

if (!clean_ptr)
	return;
```